packages/hooks/use-modal/index.ts

```diff
@@ -20,25 +19,17 @@ const closeModal = (e: KeyboardEvent) => {
   }
 }
 
-export default (
-  instance: ModalInstance,
-  visibleRef: Ref<boolean> | ComputedRef
-) => {
-  watch(
-    () => visibleRef.value,
-    (val) => {
-      if (val) {
-        modalStack.push(instance)
-      } else {
-        modalStack.splice(
-          modalStack.findIndex((modal) => modal === instance),
-          1
-        )
-      }
+export const useModal = (instance: ModalInstance, visibleRef: Ref<boolean>) => {
+  watch(visibleRef, (val) => {
+    if (val) {
+      modalStack.push(instance)
+    } else {
+      modalStack.splice(
+        modalStack.findIndex((modal) => modal === instance),
+        1
+      )
     }
-  )
+  })
 }
 
-if (!isServer) {
-  on(document, 'keydown', closeModal)
-}
+if (isClient) useEventListener(document, 'keydown', closeModal)
```

Comment:
This is not side-effect free. Shall we use some init function instead?

Reply:
Are there any side effects of useEventListener? I didn't get your point.

Reply:
@HerringtonDarkholme Can you elaborate what the side effect is here?